### ISSUE TITLE
Uninformative error / no recovery when Nix paths already exist

### ISSUE THREAD

**Issue by 9999years** [DeterminateSystems/nix-installer#213]
```
$ curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install
info: downloading installer (https://install.determinate.systems/nix/tag/v0.1.0/nix-installer-aarch64-darwin)
`nix-installer` needs to run as `root`, attempting to escalate now via `sudo`...
Password:
Sorry, try again.
Password:
Error:
   0: Planner error
   1: Error executing action
   2: Path exists `/Library/LaunchDaemons/org.nixos.darwin-store.plist`

Location:
   /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/core/src/convert/mod.rs:726

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

My install is kinda messed up, so these files exist but I don't have the `nix` tools installed etc.

I can `sudo rm` those files easy enough, but the installer should:
1. Inform users that deleting those files is safe.
2. Offer to delete those files before continuing.
3. It would be especially nice if, assuming the defaults for those files are known, the diff against those defaults could be displayed.

**Comment by 9999years** [comment]
Uh oh sisters!

```
ERROR Error saving receipt: RecordingReceipt("/nix/receipt.json", Os { code: 30, kind: ReadOnlyFilesystem, message: "Read-only file system" })
ERROR
   0: Install failure
   1: Error executing action
   2: Failed to execute command
   3: Command `"launchctl" "bootstrap" "system" "/Library/LaunchDaemons/org.nixos.darwin-store.plist"` failed status, stderr:
      /Library/LaunchDaemons/org.nixos.darwin-store.plist: service already bootstrapped
      Bootstrap failed: 37: Operation already in progress

   3:

Location:
   /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/core/src/convert/mod.rs:726
```

I just ran the Apple migration assistant from an Intel x86 machine to an M2 aarch64 machine so this is classic macOS update type shenanigans. But! These use-cases are important to support. So I will continue to document.

**Comment by 9999years** [comment]
Here it offers to revert but fails because the disk was already unmounted.

```
The following actions will be taken (`--explain` for more context):

* Remove the APFS volume `Nix Store` on `disk3`


Proceed? ([Y]es/[n]o/[e]xplain): y
 INFO Revert: Kickstart the launchctl unit `system/org.nixos.nix-daemon`
 INFO Revert: Configure Nix
 INFO Revert: Provision Nix
 INFO Revert: Create an APFS volume `Nix Store` for Nix on `disk3`
ERROR Error saving receipt: RecordingReceipt("/nix/receipt.json", Os { code: 30, kind: ReadOnlyFilesystem, message: "Read-only file system" })
Error:
   0: Error executing action
   1: Failed to execute command
   2: Command `"/usr/sbin/diskutil" "unmount" "force" "Nix Store"` failed status, stderr:
      disk3s7 was already unmounted

   2:

Location:
   /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/core/src/convert/mod.rs:726
```

**Comment by 9999years** [comment]
```
Proceed? ([Y]es/[n]o/[e]xplain): y
```
I really like the "explain" option here :)

**Comment by 9999years** [comment]
Wooo!! OK I deleted the Nix store volume in the Disk Utility and then the installer worked!

**Comment by 9999years** [comment]
I'm also a big fan of the first-class Fish shell support!!

**Comment by woodwardjd** [comment]
Related: I'm installing a new (macOS) machine, and copying data from my old machine via a `SuperDuper!` backup.  I had that external disk plugged in.  It has a couple of volumes on it, including one labeled `Nix Store` (which is the backup from the old machine, and is not mounted at `/nix`, but is mounted).  Install failed because the installer found that one, and apparently used it (maybe mounted it at `/nix`?), and it already has a bunch of stuff in it.  I guess a possible takeaway from this comment is to have this installer detect an existing volume called `Nix Store` and warn if that exists and/or already looks like an install.

**Comment by Hoverbear** [comment]
After poking at #212 we definitely need to check if a `Nix Store` named volume exists.

**Comment by Hoverbear** [comment]
I think there is still some work to do here, but we want to cut a new release to fix some bugs noted in #235. Moving this to a future milestone!

**Comment by Hoverbear** [comment]
#352 includes preliminary curing support which should resolve these!
